packages/frontend/src/hooks/styles/useTableStyles.tsx

```diff
@@ -0,0 +1,39 @@
+import { createStyles, CSSObject } from '@mantine/core';
+
+export const useTableStyles = (overrideStyles?: Record<string, CSSObject>) =>
+    createStyles((theme) => {
+        const defaultStyles: Record<string, CSSObject> = {
+            '& thead tr': {
+                backgroundColor: theme.colors.gray[0],
+            },
+
+            '& thead tr th': {
+                color: theme.colors.gray[6],
+                fontWeight: 600,
+            },
+
+            '& thead tr th, & tbody tr td': {
+                padding: '12px 20px',
+            },
+
+            '&[data-hover] tbody tr': theme.fn.hover({
+                cursor: 'pointer',
+                backgroundColor: theme.fn.rgba(theme.colors.gray[0], 0.5),
+            }),
+        };
+
+        if (overrideStyles) {
+            Object.keys(overrideStyles).forEach((key) => {
+                if (key in defaultStyles) {
+                    defaultStyles[key] = {
+                        ...defaultStyles[key],
+                        ...overrideStyles[key],
+                    };
+                } else {
+                    defaultStyles[key] = overrideStyles[key];
+                }
+            });
+        }
```

Comment:
That was my first approach actually, but if I spread `overrideStyles` inside `defaultStyles` it will not keep the content of a key that already exists. For example, if I follow that approach, for a `'& thead tr th'` and I want to override `fontWeight` and keep `color` intact, then that spreading will not extend the object and override the property I'm sending, it will override the whole key value. Hope this makes sense. 

Maybe there's a more elegant way to do this, but not sure tbh.